src/components/basic_table/basic_table.tsx

```diff
@@ -664,9 +664,9 @@ export class OuiBasicTable<T = any> extends Component<
 
       items.push({
         name: column.name,
-        key: `_data_s_${
+        key: `_data_s_${String(
           (column as OuiTableFieldDataColumnType<T>).field
-        }_${index}`,
+        )}_${index}`,
```

Comment:
out of scope but we shouldn't be using the index pas part of this, as if `columns` is re-ordered the key is regenerated. wouldn't `column.field` be sufficient?